### PR TITLE
Update release status

### PR DESCRIPTION
**Pull request by pawbu**
- using "constant" with allowed transitions
- code that finds latest metadata_all__x_x_x.json file
- setting status algorithm

### CODE REVIEW

File: datastore_version_manager/adapter/constants.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        'PENDING_DELETE'
+    ],
+    'IMMUTABLE': [
+        'RELEASED',
+        'DELETED'
+    ]
+}
+
+RELEASE_STATUS_ALLOWED_TRANSITIONS = {
+    'DRAFT': ['PENDING_RELEASE', 'PENDING_DELETE'],
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/constants.py:21]: Should we allow draft => pending_delete?
I think we should lean on the new functionality to hard delete drafts before setting a dataset to pending_delete.

Example CASE:
ADD SOME_DATASET: status draft
PENDING_DELETE SOME_DATASET: status pending_delete

in this case we should disallow setting it to pending_delete and allow "hard_delete_draft"(or whatever we end up calling this operation)

File: datastore_version_manager/adapter/datastore.py [hunk 2]
```diff
@@ -1,33 +1,36 @@
+import glob
 import json
 import os
+from distutils.version import StrictVersion
 
 
 def get_pending_operations() -> dict:
-    with open(f'{os.environ["DATASTORE_ROOT_DIR"]}/datastore/pending_operations.json', encoding = "utf-8") as f:
-            return json.load(f)
+    with open(f'{os.environ["DATASTORE_ROOT_DIR"]}/datastore/pending_operations.json', encoding="utf-8") as f:
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/datastore.py:8]: :100:

File: datastore_version_manager/adapter/datastore.py [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+    with open(f'{os.environ["DATASTORE_ROOT_DIR"]}/datastore/data_store.json', 'w', encoding="utf-8") as f:
+        return json.dump(datastore_dict, f, indent=4)
 
 
 def dataset_exists(dataset_name):
-    return(
-        os.path.isdir(f'{os.environ["DATASTORE_ROOT_DIR"]}/data/{dataset_name}') and
-        os.path.isdir(f'{os.environ["DATASTORE_ROOT_DIR"]}/metadata/{dataset_name}')
+    return (
+            os.path.isdir(f'{os.environ["DATASTORE_ROOT_DIR"]}/data/{dataset_name}') and
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/datastore.py:29]: overindented

File: datastore_version_manager/adapter/pending_operations.py [hunk 4]
```diff
@@ -1,42 +1,62 @@
...
 
-def update_release_status(dataset_name: str, release_status: str) -> None:
+def set_release_status(dataset_name: str, release_status: str, operation: str,
+                       description: str = None) -> None:
+
     pending_operations = datastore.get_pending_operations()
     pending_operations_list = pending_operations["pendingOperations"]
     try:
-        dataset = next(
+        dataset_on_pending_operations_list = next(
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/pending_operations.py:30]: might be more descriptive: dataset_**in**_pending_operations_list

File: datastore_version_manager/commands.py [hunk 5]
```diff
@@ -18,15 +19,15 @@ def new_draft_to_datastore(dataset_name: str, description: str,
         f'{built_dataset_path}/{dataset_name}__0_0.parquet',
         f'{datastore.get_data_dir_path(dataset_name)}/{dataset_name}__0_0.parquet'
     )
-    pending_operations.add_new_draft(dataset_name, operation, description)
+    pending_operations.add_new_pending_operation(dataset_name, operation, "DRAFT", description)
 
 
-def set_status_for_pending_operation(dataset_name: str, release_status: str):
-    if release_status not in ["PENDING_RELEASE", "DRAFT"]:
+def set_status(dataset_name: str, release_status: str, operation: str = None, description: str = None):
```
Comment (DanielElisenberg) [inline datastore_version_manager/commands.py:25]: :100:

**Review by DanielElisenberg — COMMENTED** [review]
Looks good :100: Just a few minor comments, and one release_status flow I'm unsure of

File: datastore_version_manager/adapter/constants.py (hunk 1 above)
Comment (pawbu) [reply datastore_version_manager/adapter/constants.py:21]: New command (hard delete)to remove dataset from pending operations if status DRAFT, PENDING_*. Removes unversioned metadata and data (if exists) from datastore.

File: datastore_version_manager/adapter/datastore.py (hunk 3 above)
Comment (pawbu) [reply datastore_version_manager/adapter/datastore.py:29]: IntelliJ keeps formatting it this way

File: datastore_version_manager/adapter/pending_operations.py (hunk 4 above)
Comment (pawbu) [reply datastore_version_manager/adapter/pending_operations.py:30]: Looks like the correct form is "on the list"
https://learnersdictionary.com/qa/is-it-correct-to-say-on-the-list-or-in-the-list

File: datastore_version_manager/adapter/constants.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    'PENDING_RELEASE': ['DRAFT'],
+    'PENDING_DELETE': ['DRAFT'],
+    'RELEASED': [],
```
Comment (vaskalan) [inline datastore_version_manager/adapter/constants.py]: Probably this is be correct:
'RELEASED': ['DRAFT']
They do that when they want CHANGE_DATA on existing dataset.

Comment (pawbu) [reply datastore_version_manager/adapter/constants.py]: It looks like you are looking at an older version of the file. Can you take a look at https://github.com/statisticsnorway/microdata-datastore-version-manager/pull/3/files

File: datastore_version_manager/adapter/constants.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+]
+
+RELEASE_STATUSES = {
+    'MUTABLE': [
+        'DRAFT',
+        'PENDING_RELEASE',
+        'PENDING_DELETE'
+    ],
+    'IMMUTABLE': [
+        'RELEASED',
```
Comment (vaskalan) [inline datastore_version_manager/adapter/constants.py:15]: Is RELEASED really immutable?

Comment (pawbu) [reply datastore_version_manager/adapter/constants.py:15]: This is from a user-perspective when they want to set a new status to a dataset. They cannot set RELEASED manually.

File: datastore_version_manager/adapter/pending_operations.py (hunk 4 above)
Comment (DanielElisenberg) [reply datastore_version_manager/adapter/pending_operations.py]: huh! Today I Learned 😄

**Pull request merged**